pkg/kubelet/dockertools/labels.go

```diff
@@ -33,216 +34,229 @@ import (
 //  * label setters and getters
 //  * label filters (maybe in the future)
 
+// Version label indicates the version of labels on a container. Whenever we make incompatible changes
+// to the labels, we should bump up the current label version.
 const (
-	kubernetesPodNameLabel                   = "io.kubernetes.pod.name"
-	kubernetesPodNamespaceLabel              = "io.kubernetes.pod.namespace"
-	kubernetesPodUIDLabel                    = "io.kubernetes.pod.uid"
-	kubernetesPodDeletionGracePeriodLabel    = "io.kubernetes.pod.deletionGracePeriod"
-	kubernetesPodTerminationGracePeriodLabel = "io.kubernetes.pod.terminationGracePeriod"
-
-	kubernetesContainerNameLabel                   = "io.kubernetes.container.name"
-	kubernetesContainerHashLabel                   = "io.kubernetes.container.hash"
-	kubernetesContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
-	kubernetesContainerTerminationMessagePathLabel = "io.kubernetes.container.terminationMessagePath"
-	kubernetesContainerPreStopHandlerLabel         = "io.kubernetes.container.preStopHandler"
+	versionLabel        = "io.kubernetes.label-version"
+	currentLabelVersion = "v1"
+)
 
-	// TODO(random-liu): Keep this for old containers, remove this when we drop support for v1.1.
-	kubernetesPodLabel = "io.kubernetes.pod.data"
+// Labels supported now
+const (
+	// Labels added from kubernetes v1.1
+	// Notice that the podNameLabel contains namespaced pod name in v1.1. However because we don't
+	// really rely on the pod name now, it should be fine.
+	// TODO(random-liu): Remove the comment above when we drop support for v1.1
+	podNameLabel       = "io.kubernetes.pod.name"
+	containerNameLabel = "io.kubernetes.container.name"
 
+	// Labels added from kubernetes v1.2
+	podNamespaceLabel              = "io.kubernetes.pod.namespace"
+	podUIDLabel                    = "io.kubernetes.pod.uid"
+	containerHashLabel             = "io.kubernetes.container.hash"
 	cadvisorPrometheusMetricsLabel = "io.cadvisor.metric.prometheus"
+
+	// Labels added from kubernetes v1.3
+	containerInfoLabel = "io.kubernetes.container.info"
 )
 
-// Container information which has been labelled on each docker container
-// TODO(random-liu): The type of Hash should be compliance with kubelet container status.
-type labelledContainerInfo struct {
-	PodName                   string
-	PodNamespace              string
-	PodUID                    types.UID
-	PodDeletionGracePeriod    *int64
-	PodTerminationGracePeriod *int64
-	Name                      string
-	Hash                      string
-	RestartCount              int
-	TerminationMessagePath    string
-	PreStopHandler            *api.Handler
+// containerMeta is the metadata we need to labelled on each docker container. Each field will be one
+// label on the container. We could also use the label to select specific containers in the future.
+// Notice that the container meta should seldom or never change.
+type containerMeta struct {
```

Comment:
@vishh Sorry, you are right. I checked the code, summary.go is using it and could be fixed.

We only need uid to select containers now, but I'm not sure whether we'll have other use cases in the future. I'm ok with group these fields into one label. I just wanted to make sure that we don't need to change this in the future, because each time we change this we have to add some code to guarantee the backward compatibility, which is really troublesome. :)